rm-datasource/src/main/java/io/seata/rm/datasource/sql/SQLVisitorFactory.java

```diff
@@ -77,7 +81,19 @@ public static SQLRecognizer get(String sql, String dbType) {
                     recognizer = new OracleSelectForUpdateRecognizer(sql, ast);
                 }
             }
-        }else {
+        } else if (JdbcConstants.POSTGRESQL.equalsIgnoreCase(dbType)) {
+            if (ast instanceof SQLInsertStatement) {
+                recognizer = new PostgresqlInsertRecognizer(sql, ast);
+            } else if (ast instanceof SQLUpdateStatement) {
+                recognizer = new PostgresqlUpdateRecognizer(sql, ast);
+            } else if (ast instanceof SQLDeleteStatement) {
+                recognizer = new PostgresqlDeleteRecognizer(sql, ast);
+            } else if (ast instanceof SQLSelectStatement) {
+                if (((SQLSelectStatement) ast).getSelect().getQueryBlock().isForUpdate()) {
+                    recognizer = new OracleSelectForUpdateRecognizer(sql, ast);
```

Comment:
please fix the bug, do not new OracleSelectForUpdateRecognizer

Reply:
fixed